packages/ember-data/lib/system/model/model.js

```diff
@@ -844,6 +817,35 @@ var Model = Ember.Object.extend(Ember.Evented, {
   },
 
   /**
+    @method _changedKeys
+    @private
+  */
+  _changedKeys: function(updates) {
+    var changedKeys = [];
+
+    if (updates && typeof updates === 'object') {
+      var original, i, value, key;
+      var keys = Ember.keys(updates);
+      var length = keys.length;
+
+      original = merge({}, this._data);
+      original = merge(original, this._attributes);
```

Comment:
if we're in the middle of a save we have values in `_inFlightAttributes` and that will win over `_attributes`.

```
var person = store.push('person', { id: 1, name: 'Tomster' });
person.set('name', 'Yehuda');
person.save();
// => request { name: 'Yehuda' }
person.set('name', 'Tom');
// <= response { name: 'Yehuda' }
```
`_data` will contain `{ name: 'Tomster' }`
`_attributes` will contain `{ name: 'Tom' }`
`_inFlightAttributes` will contain `{ name: 'Yehuda' }`

We will compare payload (`{ name: 'Yehuda' }`) to the merged original (`{ name: 'Yehuda' }`) and trigger no change.

Isn't this the correct behavior? We have `{ name: 'Tom' }` locally (since we changed that) and we still do, without triggering observer. If we want to get back to `{ name: 'Yehuda' }` we should do a rollback.